MoleculeVAE-reImplementation/dataloader.py

```diff
@@ -1,133 +1,272 @@
+        )
+        X_test, character_index_lookup_dict = one_hot_encoder(
+            X_test_s, params["normalizeSize"]
+        )
+        if params["do_prop_pred"]:
 
-        df = loadDataset(params['Data_file'], params['FromDeepchem'], params['featurizer'])
-        train_docs = df['smiles'].tolist()
-        logP = df["logP"].tolist()
-        QED = df["qed"].tolist()
-        SAS = df["SAS"].tolist()
-        train_docsy=np.stack(( logP, QED ,SAS), axis=-1)
-        X_train_s,X_test_s, Y_train, Y_test=train_test_split(train_docs,train_docsy)
+            return X_train, X_test, Y_train, Y_test, character_index_lookup_dict
+        else:
+            return X_train, X_test, character_index_lookup_dict
 
-        X_train, od = oneHotencoder(X_train_s, params['normalizeSize'])
-        X_test, od = oneHotencoder( X_test_s, params['normalizeSize'])
-        if params['do_prop_pred']:
 
-            return X_train, X_test, Y_train, Y_test,od
-        else:
-            return X_train, X_test,od
+def caculate_target_values(smiles):
+    """Calculates logP, QED and SAS for a smiles input;
+    - Inputs:
+        - smiles:
+          str. The smiles of the molecule for which we want to caluclate the values.
+    - Returns:
+        - logP:
+          water−octanol partition coefficient (logP) ref.
+        - QED
+          synthetic accessibility score (SAS)
+        - SAS
+          Estimation of Drug-likeness (QED)
+    If the molecule cannot be calculated by RDkit, returns error
+    These values calculated here sometimes disagree with the values given in the dataset from the original
+    dataset.
+    """
+    m = Chem.MolFromSmiles(smiles)
+    logp = Chem.Descriptors.MolLogP(m)
+    qed = QED.default(m)
+    sas = sascorer.calculateScore(m)
 
+    return logp, qed, sas
 
-if __name__ == '__main__':
-    parser = argparse.ArgumentParser()
-    parser.add_argument('-e', '--exp_file',
-                        help='experiment file', default='expParam.json')
-    parser.add_argument('-d', '--directory',
-                        help='exp directory', default=None)
-    args = vars(parser.parse_args())
-    if args['directory'] is not None:
-        args['exp_file'] = os.path.join(args['directory'], args['exp_file'])
 
-    params = hyperparameters.load_params(args['exp_file'])
+def extract_properties_value(smile_onehot, smile_string):
+    """Extract properties value for latent space prediction
+    - Inputs:
+        - smile_onehot: one hot vector for training.
+        - smile_string: corresponding smile string.
+    - Returns:
+        - X_final: one hot vector which has the corresponding properties value
+        - Y_values:properties value
+    """
+    X_final = []
+    Y_values = []
+    label = []
+    if os.path.exists("properties_value.json"):
+        with open("properties_value.json", "r", encoding="utf-8") as json_file:
+            properties_dic = ast.literal_eval(json.load(json_file))
```

Comment:
I don't think you should need an `eval` here and they're best avoided if possible.

Reply:
I got this from a [Japanese blog](https://note.nkmk.me/python-ast-literal-eval/) because I got some errors when I covert the string file to a dictionary object.